### PR TITLE
Alignment Dispatcher uses WorkflowBuilder

### PR DESCRIPTION
**Pull request by tmooney**
* A step on our quest to stop relying on `Workflow`!
* The old version had some UR optimizations built into it for the benefit of `PhenotypeCorrelation`.  I took those out in this conversion, but they could potentially be re-added if needed.
* This adds programmatic support for declaring an input `is_optional` in `Genome::WorkflowBuilder::DAG`s.  (The changes backing this were added in #906.)

### CODE REVIEW

**Comment by mkiwala** [comment]
@tmooney -- I plan to review this later today.  I'll see if I can get @davidlmorton to look at it too.

File: lib/perl/Genome/InstrumentData/Composite/Workflow.pm [hunk 1]
```diff
@@ -110,9 +106,8 @@ sub _run_workflow {
     Genome::Sys->disconnect_default_handles;
 
     $self->debug_message('Running workflow...');
-    my $dag = Genome::WorkflowBuilder::DAG->from_xml($workflow->save_to_xml());
     my %inputs_hash = (@$inputs); # For some reason $inputs is an array-ref
```
Comment (davidlmorton) [inline lib/perl/Genome/InstrumentData/Composite/Workflow.pm]: I kind of feel like this was a good opportunity to fix this.  It may be scope creep though.

**Comment by davidlmorton** [comment]
+1 Sorry this took so long for me to review.  I still don't fully understand some of the template business, but that is carry-over from the old version of the code so I'm not too concerned about it.

File: lib/perl/Genome/InstrumentData/Composite/Workflow.pm (hunk 1 above)
Comment (tmooney) [inline lib/perl/Genome/InstrumentData/Composite/Workflow.pm]: Good point; I'll make a follow-up PR once this one is done.  I think it'll be easier to follow as a fresh request rather than tacked on here.

**Comment by susannasiebert** [comment]
+1

**Comment by tmooney** [comment]
I've rebased to resolve the merge conflicts.  If this passes I'll see about merging it!
